File: past202004/past202004_d.fs
// https://atcoder.jp/contests/past202004-open/submissions/13484733

open System.Text.RegularExpressions

let S = stdin.ReadLine()

let a =
    [ "a"; "b"; "c"; "d"; "e"; "f"; "g"; "h"; "i"; "j"; "k"; "l"; "m"; "n"; "o"; "p"; "q"; "r"; "s"; "t"; "u"; "v"; "w"; "x"; "y"; "z"; "." ]

let pairs list1 list2 = List.allPairs list1 list2 |> List.map (fun x -> fst x + snd x)

let counter list1 =
    list1
    |> List.filter (fun x -> Regex.IsMatch(S, x))
    |> List.length

let r1 = a |> counter

let r2 = pairs a a |> counter

let r3 = pairs a (pairs a a) |> counter

(r1 + r2 + r3) |> stdout.WriteLine

File: past202004/past202004_d.cs
// https://atcoder.jp/contests/past202004-open/submissions/13482915

using System;
using System.Collections.Generic;
using System.Text.RegularExpressions;

class Program
{
    static void Main()
    {
        var a = new List<char> { 'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z', '.', };

        var S = Console.ReadLine();

        var result = 0;
        foreach (var c1 in a)
        {
            if (Regex.IsMatch(S, c1.ToString()))
            {
                result++;
            }
        }
        foreach (var c1 in a)
        {
            foreach (var c2 in a)
            {
                if (Regex.IsMatch(S, c1.ToString() + c2.ToString()))
                {
                    result++;
                }
            }
        }
        foreach (var c1 in a)
        {
            foreach (var c2 in a)
            {
                foreach (var c3 in a)
                {
                    if (Regex.IsMatch(S, c1.ToString() + c2.ToString() + c3.ToString()))
                    {
                        result++;
                    }
                }
            }
        }
        Console.WriteLine(result);
    }
}
